### PR TITLE
Add capability to handle NCEP operational observations

### PR DESCRIPTION
**Pull request by emilyhcliu**
**Enhancements made in this PR:**
(1) Added handling of sea surface temperature data
(2) Include OMP sensors for ozone data type
(3) Updated list of observation types (bufr types) for GPS and SST
(4) Added missing units for some variables
(5) Enhanced the capability of adding test fields to handle data with channel dimension

**Issues and related bug fix:**
The initial observation error values stored in @ObsError group were not correct for all-sky radiance and ozone data.

**Issues:**
- (1) The clear-sky observation error values were used for the all-sky case. While the clear-sky errors are assigned to constant for each channel globally, the all-sky errors for each channel vary at each observation location.  
- (2) The final observation error values (adjusted values) were stored in @ObsError group.  

**Bug fix:**
- Stored correct initial observation error assignment in @ObsError group for all-sky radiance and ozone data.

### CODE REVIEW

File: src/gsi-ncdiag/gsi_ncdiag.py [hunk 1]
```diff
@@ -974,7 +1008,8 @@ def toIODAobs(self, OutDir, clobber=True):
             units_values[value] = 'K'
 
         obsdata = self.df['Observation'][:]
-        obserr = self.df['error_variance'][:]
+#       obserr = self.df['error_variance'][:]
```
Comment (srherbener) [inline src/gsi-ncdiag/gsi_ncdiag.py]: Please remove commented code. Thanks

File: src/gsi-ncdiag/gsi_ncdiag.py [hunk 2]
```diff
@@ -1030,7 +1081,8 @@ def toIODAobs(self, OutDir, clobber=True):
                 continue
             obsdatasub = obsdata[idx]
             obsdatasub[obsdatasub > 9e5] = nc.default_fillvals['f4']
-            obserrsub = np.full(nlocs, obserr[c])
+#           obserrsub = np.full(nlocs, obserr[c])
```
Comment (srherbener) [inline src/gsi-ncdiag/gsi_ncdiag.py]: Please remove commented code. Thanks

File: src/gsi-ncdiag/gsi_ncdiag.py [hunk 3]
```diff
@@ -1439,9 +1491,11 @@ def toIODAobs(self, OutDir, clobber=True):
         varDict[vname]['qcKey'] = vname, writer.OqcName()
 
         obsdata = self.df['Observation'][:]
-        tmp = self.df['Inverse_Observation_Error'][:]
+#       tmp = self.df['Inverse_Observation_Error'][:]
```
Comment (srherbener) [inline src/gsi-ncdiag/gsi_ncdiag.py]: Please remove commented code. Thanks

File: src/gsi-ncdiag/gsi_ncdiag.py [hunk 4]
```diff
@@ -1439,9 +1491,11 @@ def toIODAobs(self, OutDir, clobber=True):
...
+        tmp = self.df['Input_Observation_Error'][:]
         tmp[tmp < 9e-12] = 0
-        obserr = 1.0 / tmp
+#       obserr = 1.0 / tmp
```
Comment (srherbener) [inline src/gsi-ncdiag/gsi_ncdiag.py]: Please remove commented code. Thanks

**Comment by emilyhcliu** [comment]
@srherbener Comment lines removed.

**Comment by emilyhcliu** [comment]
@CoryMartin-NOAA I was about to write you an e-mail about the request to review this PR.  Thank you.

**Comment by emilyhcliu** [comment]
@srherbener Thanks for your review.

**Pull request merged**